### ISSUE TITLE
Strange behaviors when server side rendering using JS engine pool

### ISSUE THREAD

**Issue by fremler** [reactjs/React.NET#1230]
### Please verify these steps before filing an issue, and check them off as you go
- [x] The relevant native JavascriptEngineSwitcher library packages are installed (such as `JavaScriptEngineSwitcher.V8.Native.win-x64`)
- [x] The VC++ 2017 runtime is installed
- [x] The value of `SetUseReact` and `SetUseBabel` is correct in `ReactConfig.cs` or `Startup.cs`
- [x] I've looked at the sample projects in this repo to verify that my app is configured correctly

### I'm using these library versions:
- `ReactJS.NET`: 5.2.11
- `JavaScriptEngineSwitcher`: 3.3.0
- `react` and `react-dom`: (N/A if using bundled react, or version number)
- `webpack`: (N/A if using bundled react)
- `node`: (N/A if using bundled react)

### Runtime environment:
- OS: Windows 10 x64
- .NET Framework or .NET Core Version: .NET Framework 4.8

Maybe I'm missing something here but I just want to raise the question anyway. I'm experiencing some strange behaviors when server side rendering. I'm using the Chakra Core JS engine and it seems like when reusing the engines using `SetReuseJavaScriptEngines` the method call JavaScriptEngineSwitcher.ChakraCore.ScriptDispatcher.StartThread() take a lot longer to execute. 

SetReuseJavaScriptEngines(false)
![SetReuseJavaScriptEngines(false)](https://user-images.githubusercontent.com/20284395/100120812-aab6ec00-2e78-11eb-87ed-38d15f66ea75.png)

SetReuseJavaScriptEngines(true)
![SetReuseJavaScriptEngines(true)](https://user-images.githubusercontent.com/20284395/100120832-b1ddfa00-2e78-11eb-96cf-1fad984d5a2d.png)

Can someone explain to me what's going on? I've tried looking through the docs but can't find any further explanation to how the engine pool work other the [this old ticket](https://github.com/reactjs/React.NET/issues/13) when it was implemented. The reusing of the JS engines over request is supposed to improve performance, but here it looks like it have negative impact on performance.

**Comment by xecrets** [comment]
Hi, I think this is a misinterpretation of the situation. A ScriptDispatcher instance is allocated per JS Engine. The ScriptDispatcher essentially instantiates a new thread with the thread proc being 'StartThread' (a better name would probably have been something like 'ScriptDispatcherLoop' or similar).

The method 'StartThread' does not start a thread - it sits in a loop waiting for scripts to execute in it's own thread.

So, when you re-use the engine, this thread becomes long-lived since the engine is re-used. What you're seeing is not something taking a long time - you're seeing the (positive) effect of the engine being still alive in the pool, and it's ScriptDispatcher sitting there waiting for something to do.

In summary - all seems well and is behaving as expected.

**Comment by fremler** [comment]
@xecrets Thank you very much for taking the time to explain in detail.
